src/tools/server.ml

```diff
@@ -209,6 +209,7 @@ let make_server_thread
         Lwt.catch
           (fun () ->
            Lwt_unix.accept listening_socket >>= fun (plain_fd, cl_socket_address) ->
+           Lwt_unix.setsockopt listening_socket Unix.SO_KEEPALIVE true;
```

Comment:
Ah, it wasn't clear to me yet that these values could be overridden on socket level.

It would indeed be a good idea to also make these configurable...

```
 The procedures involving keepalive use three user-driven variables:

tcp_keepalive_time
    the interval between the last data packet sent (simple ACKs are not considered data) and the first keepalive probe; after the connection is marked to need keepalive, this counter is not used any further 

tcp_keepalive_intvl
    the interval between subsequential keepalive probes, regardless of what the connection has exchanged in the meantime 

tcp_keepalive_probes
    the number of unacknowledged probes to send before considering the connection dead and notifying the application layer 
```

Any sane values you would recommend?
We could go for e.g. CNT=3, IDLE=20, INTVL=20?